galsim/chromatic.py

```diff
@@ -211,17 +218,18 @@ def _fiducial_profile(self, bandpass):
 
     def __eq__(self, other):
         return (isinstance(other, galsim.ChromaticObject) and
-                self.obj == other.obj)
+                hasattr(other, '_obj') and  # not all ChromaticObjects have an _obj attribute.
```

Comment:
This feels a little weird to me.  Two `ChromaticObject`s can only compare equal if they both have `_obj` attrs (and they're equal).  As it turns out, I think only `ChromaticObject` proper and `ChromaticFourierSqrt` can trigger this comparison, since the other subclasses override with their own `__eq__`.  Since both `ChromaticObject` and `ChromaticFourierSqrt` both have `_obj` defined, this works (actually, I suppose this means the`hasattr` check isn't actually needed).  But if we ever defined another subclass without a `._obj` and without it's own `__eq__`, this might inappropriately fail.  Not sure if this really needs attention or not, but I thought I'd flag it.

Reply:
I think it is required for e.g. `ChromaticObject() == ChromaticConvolution()`.

Reply:
Actually, I guess Python's decision tree about how to call `__eq__` is [more complicated than I thought](https://stackoverflow.com/questions/3588776/how-is-eq-handled-in-python-and-in-what-order), so I guess it wasn't actually necessary.  Still it seems like an appropriate thing to include here.